src/lib/snack-bar/snack-bar-config.ts

```diff
@@ -35,7 +35,13 @@ export class MatSnackBarConfig {
   duration?: number = 0;
 
   /** Extra CSS classes to be added to the snack bar container. */
-  extraClasses?: string[];
+  panelClass?: string[];
```

Comment:
I have two concerns with this,

1. This adds the class to the snackbar container, not the overlay panel. So it's slightly misleading and inconsistent with dialogs in that regard.
2. More importantly, I think `panelClass` implies that a single string would be sufficient. Since `string[]` is req'd, the snackbar container will convert `panelClass: "my-class"` to 

    ```
    <snack-bar-container class="m y - c l a s">...</snack-bar-container>
    ```

    I think the `MatSnackBar#_createOverlay` could use the overlay-ref logic to automatically handle this .

I know this PR is trying to only change the names, but I think this name change is misrepresenting unless it's also met with a change in functionality.

Reply:
Agreed; I didn't realize snackbar was `string[]` only right now. @amcdnl let's _just_ change this one to `string | string[]` as well